### ISSUE TITLE
FormDef::deleteRepeat  causes NullPointerException -- erroneous multiplicity adjustments

### ISSUE THREAD

**Issue by getodk-bot** [getodk/javarosa#142]
Originally reported by: **Mitch Sundt (Bitbucket: [m.sundt](https://bitbucket.org/m.sundt), GitHub: Unknown)**

----------------------------------------

This exists in javarosa and javarosa-pending.

https://bitbucket.org/javarosa/javarosa/src/8acbd7900f03/core/src/org/javarosa/core/model/FormDef.java

lines 293-299:
                // update multiplicities of other child nodes
                for (int i = 0; i < parentElement.getNumChildren(); i++) {
                        TreeElement child = parentElement.getChildAt(i);
                        if (child.getMult() > childMult) {
                                child.setMult(child.getMult() - 1);
                        }
                }

There should be an additional guard condition on the if:

			if (child.getName().equals(deleteElement.getName()) 
                            && child.getMult() > childMult) {
				child.setMult(child.getMult() - 1);
			}

Attached is an example form.  If you add 2 owners, then add 2 pets, and then delete the pets, the symptom is that the multiplicities for the 2 owners gets messed up (Force Close in ODK Collect 1.2).

Basically, the deletion of the pets causes an adjustment to the multiplicities of the owners, leading to the 2 owners both having '0' multiplicity.

----------------------------------------
- Bitbucket: https://bitbucket.org/javarosa/javarosa/issue/66

**Comment by lognaturel** [comment]
> 		if (child.getName().equals(deleteElement.getName()) 
                        && child.getMult() > childMult) {
			child.setMult(child.getMult() - 1);
		}

This exists on `master` and we haven't heard of issues on repeat deletion.
